ArduPlane/commands_logic.cpp

```diff
@@ -718,10 +714,15 @@ bool Plane::verify_loiter_to_alt()
     update_loiter(mission.get_current_nav_cmd().p1);
 
     //has target altitude been reached?
-    if (condition_value != 0) {
+    // condition_value == 1 means alt is ignored: we've either reached it or it's not valid
+    if (condition_value == 0) {
         // primary goal, loiter alt
-        if (labs(condition_value - current_loc.alt) < 500 && loiter.sum_cd > 1) {
+        if (labs(loiter.sum_cd) > 1 && (loiter.reached_target_alt || loiter.unable_to_acheive_target_alt)) {
             // primary goal completed, initialize secondary heading goal
+            if (loiter.unable_to_acheive_target_alt) {
+                gcs_send_text(MAV_SEVERITY_WARNING,"Verify nav: LOITER alt was stuck");
```

Comment:
"Loiter to altitude has been detected as stuck at %d meters". The verify nav stuff is a developer info only, a nicer user string can still be searched in the code base back to here easily. (Basically we shouldn't have the verify nav as thats confusing to a user, and LOITER in caps lock is just messy formatting.